torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1816,7 +1849,7 @@ def _find(self, a: "sympy.Symbol") -> "sympy.Expr":
             return a
         res = self.replacements[a]
         cur_replace = {s: self._find(s) for s in res.free_symbols}
-        self.replacements[a] = self.replacements[a].xreplace(cur_replace)
+        self._set_replacement(a, self.replacements[a].xreplace(cur_replace))
```

Comment:
would like to check if the extra equality test for every union find replacement is a performance problem

Reply:
I put it under the condition for printing, which includes a constant check. Couldn't get rid of it entirely (otherwise will get a duplicate log message); added a comment.